### PR TITLE
feat(api-key): support api_key/api_secret authorize

### PR DESCRIPTION
**Pull request by zhongwencool**

### CODE REVIEW

File: apps/emqx/src/emqx_schema.erl [hunk 1]
```diff
@@ -1374,6 +1383,16 @@ to_comma_separated_list(Str) ->
 to_comma_separated_atoms(Str) ->
     {ok, lists:map(fun to_atom/1, string:tokens(Str, ", "))}.
 
+rfc3339_to_system_time(DateTime) ->
+    try
+        {ok, calendar:rfc3339_to_system_time(DateTime, [{unit, second}])}
+    catch error: {badmatch, _} ->
+        {error, ["The rfc3339 specification not satisfied"]}
```
Comment (zmstone) [inline apps/emqx/src/emqx_schema.erl:1390]: why return a list of strings here ?
```suggestion
        {error, ["bad_rfc3339_timestamp"]}
```

File: apps/emqx/src/emqx_schema.erl [hunk 2]
```diff
@@ -1374,6 +1383,16 @@ to_comma_separated_list(Str) ->
 to_comma_separated_atoms(Str) ->
     {ok, lists:map(fun to_atom/1, string:tokens(Str, ", "))}.
 
+rfc3339_to_system_time(DateTime) ->
+    try
+        {ok, calendar:rfc3339_to_system_time(DateTime, [{unit, second}])}
+    catch error: {badmatch, _} ->
```
Comment (zmstone) [inline apps/emqx/src/emqx_schema.erl:1389]:
```suggestion
    catch error: _ ->
```

File: apps/emqx/src/emqx_schema.erl (hunk 1 above)
Comment (zmstone) [reply apps/emqx/src/emqx_schema.erl:1390]: or maybe `{error, bad_rfc3339_timestamp}`

Comment (zhongwencool) [reply apps/emqx/src/emqx_schema.erl]: good catch! `{error, bad_rfc3339_timestamp}` is better.

**Review by zmstone — APPROVED** [review]

**Pull request merged**